cylc/flow/network/resolvers.py

```diff
@@ -34,6 +35,10 @@
     NodesEdges, PROXY_NODES, SUB_RESOLVERS, parse_node_id, sort_elements
 )
 
+if TYPE_CHECKING:
+    from cylc.flow.scheduler import Scheduler
```

Comment:
Avoid unnecessary imports mainly, this will only import the Scheduler when mypy is run

Reply:
I'm guessing we can't do:

```
self.schd: 'cylc.flow.scheduler.Scheduler' = schd
```

In order for a `Scheduler` object to have been passed to the `Resolvers` instance `cylc.flow.scheduler` must already be loaded in memory, even so it would be annoying if we had to write imports like this everywhere to keep mypy happy.

Reply:
Imho writing out the full path like that would be more annoying than doing `if TYPE_CHECKING... import...`. Anyway, hopefully the usefulness of mypy outweighs its annoyances.